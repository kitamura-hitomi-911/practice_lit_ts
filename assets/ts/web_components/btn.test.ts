import { LitElement } from "lit";
test("btn", () => {
  const buttonElement = window.document.createElement("ss-btn") as LitElement;
  console.log("★", buttonElement.shadowRoot?.querySelector("button"));
  // Set up our document body
  document.body.innerHTML = "<ss-btn>ボタンラベル</ss-btn>";
  console.log(document.body.innerHTML);
  const tgtElm = document.getElementsByTagName("ss-btn");
  const btnElm = tgtElm[0].shadowRoot?.querySelector("button");
  console.log(tgtElm[0].shadowRoot?.textContent, btnElm?.textContent);
  console.log(tgtElm[0].textContent);
  expect(tgtElm[0]).toMatchSnapshot();

  // Tell the fetchCurrentUser mock function to automatically invoke
  // its callback with some data
  expect("aa").toBe("aa");
});
