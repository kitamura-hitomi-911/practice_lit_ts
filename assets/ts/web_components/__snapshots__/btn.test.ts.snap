// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`btn 1`] = `
<ss-btn>
  ボタンラベル
</ss-btn>
`;
